Read more here: http://www.ianbicking.org/blog/2007/09/re-raising-exceptions.html

*raise* in Python is used for two primary reasons:
* raising your own errors
* re-raising the errors 

Re-raising block should include
* reverting
* dealing with tracetrack (e.g. printing it)
* probably printing specific error message

For examples see the link above
